### PR TITLE
AP-72 linpeas export

### PR DESCRIPTION
**Pull request by RobinSiep**
## Code R Ticket
_Please provide a link to the Code R Jira ticket if applicable_

<https://partypeak.atlassian.net/browse/AP-72>

## Description
Adds a fab allowing the user to export their linpeas report when the current run has finished.

image

image

## Dependencies
_Are there any dependencies in relation to pull requests in other Apollo repositories? If so please provide links to these pull requests_

* https://github.com/thecoderstudio/apollo/pull/42

## Additional notes
Nope

### CODE REVIEW

File: src/components/actions/ExportLinpeas.jsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import styled from 'styled-components';
+import Switch from 'react-switch';
+import axios from 'axios';
+import PropTypes from 'prop-types';
+import { StatusCodes } from 'http-status-codes';
+import { exportLinpeasSchema } from '../../validation/linpeas';
+import { handleHTTPResponse } from '../../actions/error';
+import { downloadResponse } from '../../util/http';
+import { parseHTTPErrors } from '../../util/parser';
+import media from '../../util/media';
```
Comment (rikvanderwerf) [inline src/components/actions/ExportLinpeas.jsx]: unused

File: src/components/actions/ExportLinpeas.jsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  export(values, { setErrors }) {
+    const { agent } = this.props;
+    axios.get(
+      `${process.env.APOLLO_HTTP_URL}agent/${agent.id}/action/linpeas/export`,
+      {
+        withCredentials: true,
+        params: {
+          ansi: this.state.ansi,
+          filename: values.filename ? values.filename : null
+        },
```
Comment (rikvanderwerf) [inline src/components/actions/ExportLinpeas.jsx]:
```suggestion
        }
```

File: src/util/http.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+function getFilenameFromHeaders(headers) {
+    try {
+      return headers['content-disposition'].split('filename=')[1];
+    } catch (error) {
+      if (error instanceof TypeError) {
+        return null;
+      }
+      throw error;
+    }
```
Comment (rikvanderwerf) [inline src/util/http.js]:
```suggestion
  try {
    return headers['content-disposition'].split('filename=')[1];
  } catch (error) {
    if (error instanceof TypeError) {
      return null;
    }
    throw error;
  }
```

File: tests/components/modals/FormModal.test.jsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    <FormModal
+        title="Test"
+        primaryActionTitle="Create"
+        initialValues={{ test: '' }}
+        onClose={onClose}
+        onSubmit={onSubmit}>
+        {(values, errors, handleChange) => {
+          expect(values.test).toEqual('');
+          return <p>Test form</p>;
+        }}
```
Comment (rikvanderwerf) [inline tests/components/modals/FormModal.test.jsx]:
```suggestion
      title="Test"
      primaryActionTitle="Create"
      initialValues={{ test: '' }}
      onClose={onClose}
      onSubmit={onSubmit}>
      {(values, errors, handleChange) => {
        expect(values.test).toEqual('');
        return <p>Test form</p>;
      }}
```

**Review by rikvanderwerf — CHANGES_REQUESTED** [review]
Also I can't click the button everywhere.
IMG_0101
IMG_0100

**Comment by RobinSiep** [comment]
> Also I can't click the button everywhere.

Did you check with the latest changes?

**Pull request closed without merging**